src/components/project/handlers/apply-approved-changeset-to-project.handler.ts

```diff
@@ -35,17 +44,61 @@ export class ApplyApprovedChangesetToProject
       }
       const { id, createdAt, type, financialReportPeriod, ...actualChanges } =
         changes;
-      await this.projectService.update(
-        {
-          id,
-          ...actualChanges,
-          ...(financialReportPeriod ? { financialReportPeriod } : {}),
-        },
-        event.session,
-        undefined,
-        false
+
+      await Promise.all(
+        keys(actualChanges).map(async (key) => {
+          const query = this.db
+            .query()
+            .match(node('node', 'Project', { id }))
+            // Apply previous label to active prop
+            .apply(addPreviousLabel(key, changesetId))
+            // Deactivate active prop
+            .apply(
+              deactivateProperty({
+                key: key as keyof Omit<
+                  DbChanges<TranslationProject | InternshipProject>,
+                  'canDelete'
+                >,
+                resource:
+                  type === ProjectType.Translation
+                    ? TranslationProject
+                    : InternshipProject,
+              })
+            )
+            // Set changed prop to active
+            .apply(activeChangedProp(key, changesetId))
+            .return('node');
+
+          await query.run();
+        })
       );
 
+      // TODO
+      // const query = this.db
+      //   .query()
+      //   .match([
+      //     node('node', 'Project'),
+      //     relation('out', '', 'changeset', { active: true }),
+      //     node('changeset', 'Changeset', { id: changesetId }),
+      //   ])
+      //   .apply(matchProps({ changeset: changesetId, optional: true }))
+      //   .with('props')
+      //   .forEach('prop', 'props', (prop) =>
```

Comment:
@CarsonF Tried to use your `forEach` query but not sure how to get `key`.

Reply:
`matchProps` composes the a prop list to an object. Which for most cases is what we want.
Here however we want a list. I would just do [the match](https://github.com/SeedCompany/cord-api-v3/blob/02087ef8f091f5253f2404da3a7aee5570564e07/src/core/database/query/matching.ts#L81-L96) yourself here.

That would make each row be an individual property, which means you wouldn't need the foreach.